child_compassion/models/child_compassion.py

```diff
@@ -537,6 +537,32 @@ def fetch_translations(self):
         self._fetch_translations(self.env.ref("child_compassion.beneficiaries_details"))
         return self.edit_translations()
 
+    @api.model
+    def update_all_children_description(self):
+        children = self.env["compassion.child"].sudo().search([])
+        infos = ""
+        for child in children:
+            error = child.update_child_descriptions()
+            if len(error) > 0:
+                infos += "\n%s" % error
+        if len(infos) > 0:
+            raise UserError(infos)
```

Comment:
Maybe launching a background job for each child will avoid a very long process being stuck. The error is returned and will therefore be visible in the job result. Priority 50 is for allowing other jobs to be executed before those.
```suggestion
            child.with_delay(priority=50).update_child_descriptions()
```